### PR TITLE
preempt/cancel test for time behavior, spin pluguin

### PR DESCRIPTION
**Pull request by stevedanomodolor**
---

## Basic Info

| Info | Please fill out this column |
| ------ | ----------- |
| Ticket(s) this addresses   | (add tickets here ) |
| Primary OS tested on | (Ubuntu) |
| Robotic platform tested on | (Simulation) |

---

## Description of contribution in a few bullet points
* include preempt/cancel test for time behavior, in the spin plugin

## Description of documentation updates required from your changes

---

#### For Maintainers: 
- [x] Check that any new parameters added are updated in navigation.ros.org
- [x] Check that any significant change is added to the migration guide
- [x] Check that any new features **OR** changes to existing behaviors are reflected in the tuning guide
- [x] Check that any new functions have Doxygen added
- [x] Check that any new features have test coverage
- [x] Check that any new plugins is added to the plugins page
- [x] If BT Node, Additionally: add to BT's XML index of nodes for groot, BT package's readme table, and BT library lists

### CODE REVIEW

**Comment by SteveMacenski** [comment]
Make sure to run `ament_cpplint` and `ament_uncrustify` locally - there are a number of linting failures

File: nav2_system_tests/src/behaviors/spin/spin_behavior_tester.hpp [hunk 1]
```diff
@@ -57,7 +57,9 @@ class SpinBehaviorTester
   // Runs a single test with given target yaw
   bool defaultSpinBehaviorTest(
     float target_yaw,
-    double tolerance = 0.1);
+    double tolerance = 0.1,
+    bool wait_action = true,
```
Comment (SteveMacenski) [inline nav2_system_tests/src/behaviors/spin/spin_behavior_tester.hpp:61]: Can we rename this to `nonblocking_action` to be more clear for myself when I read this in a few years :laughing:

**Review by SteveMacenski — CHANGES_REQUESTED** [review]
Other than CI failures due to linting, this looks good to me!

**Comment by SteveMacenski** [comment]
The tests you added are failing in CI

**Comment by stevedanomodolor** [comment]
> The tests you added are failing in CI

I just noticed the reason why, I actually thought that preempt returned true, but based on the code it returns false. I will add a comment for future reference. https://github.com/ros-planning/navigation2/blob/a392a17809178a1b1fd2bdbaca6204b579c5a81a/nav2_behaviors/include/nav2_behaviors/timed_behavior.hpp#L223

**Comment by stevedanomodolor** [comment]
Ci issue fixed

**Comment by SteveMacenski** [comment]
mhm, the code coverage doesn't show this as being covered https://app.codecov.io/gh/ros-planning/navigation2/blob/tests%2Frecovery_behaviors_cancel_preempt/nav2_behaviors/include/nav2_behaviors/timed_behavior.hpp

Can you check the logs of your tests? Do you see those logs in the preempt / cancel condition being listed on the screen? I suspect something's not quite right here

**Comment by stevedanomodolor** [comment]
Actually it does. It is strange that it does not appear in the coverage
```
24: [test_spin_behavior_fake_node-6] [INFO] [1669705889.232778337] [spin_behavior_test]: Spin action server is ready
24: [test_spin_behavior_fake_node-6] [ RUN      ] SpinBehaviorTestFixture.testSpinPreemption
24: [test_spin_behavior_fake_node-6] [INFO] [1669705894.234137257] [spin_behavior_test]: Found current robot pose
24: [test_spin_behavior_fake_node-6] [INFO] [1669705894.234233839] [spin_behavior_test]: Init Yaw is 0.000000
24: [test_spin_behavior_fake_node-6] [INFO] [1669705894.234259843] [spin_behavior_test]: Before sending goal
24: [behavior_server-4] [INFO] [1669705894.335794698] [behavior_server]: Running spin
24: [behavior_server-4] [INFO] [1669705894.336128320] [behavior_server]: Turning 9.42 for spin behavior.
24: [test_spin_behavior_fake_node-6] [INFO] [1669705901.336675454] [spin_behavior_test]: Found current robot pose
24: [test_spin_behavior_fake_node-6] [INFO] [1669705901.336752952] [spin_behavior_test]: Init Yaw is 0.000000
24: [test_spin_behavior_fake_node-6] [INFO] [1669705901.336777119] [spin_behavior_test]: Before sending goal
24: [test_spin_behavior_fake_node-6] [INFO] [1669705901.438115551] [spin_behavior_test]: Waiting for result
24: [behavior_server-4] [ERROR] [1669705901.536428595] [behavior_server]: Received a preemption request for spin, however feature is currently not implemented. Aborting and stopping.
24: [behavior_server-4] [WARN] [1669705901.536627725] [behavior_server]: [spin] [ActionServer] Aborting handle.
24: [behavior_server-4] [INFO] [1669705901.536833768] [behavior_server]: Running spin
24: [behavior_server-4] [INFO] [1669705901.536939212] [behavior_server]: Turning 12.57 for spin behavior.
24: [test_spin_behavior_fake_node-6] [INFO] [1669705902.438720147] [spin_behavior_test]: target_yaw 12.566371
24: [behavior_server-4] [WARN] [1669705902.537220580] [behavior_server]: Collision Ahead - Exiting Spin
24: [behavior_server-4] [WARN] [1669705902.537247797] [behavior_server]: spin failed
24: [behavior_server-4] [WARN] [1669705902.537255032] [behavior_server]: [spin] [ActionServer] Aborting handle.
24: [test_spin_behavior_fake_node-6] [INFO] [1669705903.146573041] [spin_behavior_test]: After sending goal
24: [test_spin_behavior_fake_node-6] [ERROR] [1669705903.147398285] [spin_behavior_test]: Goal was aborted
24: [test_spin_behavior_fake_node-6] [       OK ] SpinBehaviorTestFixture.testSpinPreemption (13914 ms)
24: [test_spin_behavior_fake_node-6] [ RUN      ] SpinBehaviorTestFixture.testSpinCancel
24: [test_spin_behavior_fake_node-6] [INFO] [1669705908.148230933] [spin_behavior_test]: Found current robot pose
24: [test_spin_behavior_fake_node-6] [INFO] [1669705908.148308387] [spin_behavior_test]: Init Yaw is 0.000000
24: [test_spin_behavior_fake_node-6] [INFO] [1669705908.148332576] [spin_behavior_test]: Before sending goal
24: [behavior_server-4] [INFO] [1669705908.249599161] [behavior_server]: Running spin
24: [behavior_server-4] [INFO] [1669705908.249771621] [behavior_server]: Turning 12.57 for spin behavior.
24: [test_spin_behavior_fake_node-6] [INFO] [1669705910.250820766] [spin_behavior_test]: Waiting for result
24: [behavior_server-4] [INFO] [1669705910.350111266] [behavior_server]: Canceling spin
24: [behavior_server-4] [WARN] [1669705910.350260623] [behavior_server]: [spin] [ActionServer] Client requested to cancel the goal. Cancelling.
24: [test_spin_behavior_fake_node-6] [INFO] [1669705911.251422358] [spin_behavior_test]: target_yaw 12.566371
24: [test_spin_behavior_fake_node-6] [INFO] [1669705911.967656375] [spin_behavior_test]: After sending goal
24: [test_spin_behavior_fake_node-6] [ERROR] [1669705911.968027722] [spin_behavior_test]: Goal was canceled
24: [test_spin_behavior_fake_node-6] [       OK ] SpinBehaviorTestFixture.testSpinCancel (8821 ms)
24: [test_spin_behavior_fake_node-6] [----------] 2 tests from SpinBehaviorTestFixture (22735 ms total)
24: [test_spin_behavior_fake_node-6] 
24: [test_spin_behavior_fake_node-6] [----------] 7 tests from SpinRecoveryTests/SpinBehaviorTestFixture
24: [test_spin_behavior_fake_node-6] [INFO] [1669705921.988706070] [spin_behavior_test]: Spin action server is ready
24: [test_spin_behavior_fake_node-6] [ RUN      ] SpinRecoveryTests/SpinBehaviorTestFixture.testSpinRecovery/0523599_0100000
24: [test_spin_behavior_fake_node-6] [INFO] [1669705926.989652924] [spin_behavior_test]: Found current robot pose
24: [test_spin_behavior_fake_node-6] [INFO] [1669705926.989746530] [spin_behavior_test]: Init Yaw is 0.000000
24: [test_spin_behavior_fake_node-6] [INFO] [1669705926.989783649] [spin_behavior_test]: Before sending goal
24: [behavior_server-4] [INFO] [1669705927.091160685] [behavior_server]: Running spin
24: [behavior_server-4] [INFO] [1669705927.091376516] [behavior_server]: Turning -0.52 for spin behavior.
24: [test_spin_behavior_fake_node-6] [INFO] [1669705927.091251572] [spin_behavior_test]: Waiting for result
24: [test_spin_behavior_fake_node-6] [INFO] [1669705928.091982664] [spin_behavior_test]: target_yaw -0.523599
24: [test_spin_behavior_fake_node-6] [INFO] [1669705928.120496931] [spin_behavior_test]: After sending goal
24: [behavior_server-4] [INFO] [1669705928.191911727] [behavior_server]: spin completed successfully
24: [test_spin_behavior_fake_node-6] [INFO] [1669705928.192400126] [spin_behavior_test]: result received
24: [test_spin_behavior_fake_node-6] [       OK ] SpinRecoveryTests/SpinBehaviorTestFixture.testSpinRecovery/0523599_0100000 (6204 ms)
24: [test_spin_behavior_fake_node-6] [ RUN      ] SpinRecoveryTests/SpinBehaviorTestFixture.testSpinRecovery/0785398_0100000
24: [test_spin_behavior_fake_node-6] [INFO] [1669705933.193397322] [spin_behavior_test]: Found current robot pose
24: [test_spin_behavior_fake_node-6] [INFO] [1669705933.193477657] [spin_behavior_test]: Init Yaw is 0.000000
24: [test_spin_behavior_fake_node-6] [INFO] [1669705933.193501288] [spin_behavior_test]: Before sending goal
24: [behavior_server-4] [INFO] [1669705933.294702682] [behavior_server]: Running spin
24: [behavior_server-4] [INFO] [1669705933.294895337] [behavior_server]: Turning 0.79 for spin behavior.
24: [test_spin_behavior_fake_node-6] [INFO] [1669705933.294870865] [spin_behavior_test]: Waiting for result
24: [test_spin_behavior_fake_node-6] [INFO] [1669705934.295568028] [spin_behavior_test]: target_yaw 0.785398
24: [test_spin_behavior_fake_node-6] [INFO] [1669705934.336243979] [spin_behavior_test]: After sending goal
24: [behavior_server-4] [INFO] [1669705934.395427608] [behavior_server]: spin completed successfully
24: [test_spin_behavior_fake_node-6] [INFO] [1669705934.395867285] [spin_behavior_test]: result received
24: [test_spin_behavior_fake_node-6] [       OK ] SpinRecoveryTests/SpinBehaviorTestFixture.testSpinRecovery/0785398_0100000 (6204 ms)
24: [test_spin_behavior_fake_node-6] [ RUN      ] SpinRecoveryTests/SpinBehaviorTestFixture.testSpinRecovery/1570796_0100000
24: [test_spin_behavior_fake_node-6] [INFO] [1669705939.396698613] [spin_behavior_test]: Found current robot pose
24: [test_spin_behavior_fake_node-6] [INFO] [1669705939.396775396] [spin_behavior_test]: Init Yaw is 0.000000
24: [test_spin_behavior_fake_node-6] [INFO] [1669705939.396798672] [spin_behavior_test]: Before sending goal
24: [behavior_server-4] [INFO] [1669705939.497967735] [behavior_server]: Running spin
24: [behavior_server-4] [INFO] [1669705939.498199329] [behavior_server]: Turning -1.57 for spin behavior.
24: [test_spin_behavior_fake_node-6] [INFO] [1669705939.498102353] [spin_behavior_test]: Waiting for result
24: [test_spin_behavior_fake_node-6] [INFO] [1669705940.498610263] [spin_behavior_test]: target_yaw -1.570796
24: [test_spin_behavior_fake_node-6] [INFO] [1669705940.585189994] [spin_behavior_test]: After sending goal
24: [behavior_server-4] [INFO] [1669705940.598677738] [behavior_server]: spin completed successfully
24: [test_spin_behavior_fake_node-6] [INFO] [1669705940.599123775] [spin_behavior_test]: result received
24: [test_spin_behavior_fake_node-6] [       OK ] SpinRecoveryTests/SpinBehaviorTestFixture.testSpinRecovery/1570796_0100000 (6203 ms)
24: [test_spin_behavior_fake_node-6] [ RUN      ] SpinRecoveryTests/SpinBehaviorTestFixture.testSpinRecovery/3141593_0100000
24: [test_spin_behavior_fake_node-6] [INFO] [1669705945.600035889] [spin_behavior_test]: Found current robot pose
24: [test_spin_behavior_fake_node-6] [INFO] [1669705945.600116375] [spin_behavior_test]: Init Yaw is 0.000000
24: [test_spin_behavior_fake_node-6] [INFO] [1669705945.600140624] [spin_behavior_test]: Before sending goal
24: [behavior_server-4] [INFO] [1669705945.700808107] [behavior_server]: Running spin
24: [behavior_server-4] [INFO] [1669705945.700881246] [behavior_server]: Turning 3.14 for spin behavior.
24: [test_spin_behavior_fake_node-6] [INFO] [1669705945.700856079] [spin_behavior_test]: Waiting for result
24: [test_spin_behavior_fake_node-6] [INFO] [1669705946.701530741] [spin_behavior_test]: target_yaw 3.141593
24: [behavior_server-4] [WARN] [1669705946.801308828] [behavior_server]: Collision Ahead - Exiting Spin
24: [behavior_server-4] [WARN] [1669705946.801382439] [behavior_server]: spin failed
24: [behavior_server-4] [WARN] [1669705946.801410294] [behavior_server]: [spin] [ActionServer] Aborting handle.
24: [test_spin_behavior_fake_node-6] [INFO] [1669705946.871598474] [spin_behavior_test]: After sending goal
24: [test_spin_behavior_fake_node-6] [ERROR] [1669705946.872180306] [spin_behavior_test]: Goal was aborted
24: [test_spin_behavior_fake_node-6] [INFO] [1669705951.872879248] [spin_behavior_test]: Found current robot pose
24: [test_spin_behavior_fake_node-6] [INFO] [1669705951.872959295] [spin_behavior_test]: Init Yaw is 0.000000
24: [test_spin_behavior_fake_node-6] [INFO] [1669705951.872982347] [spin_behavior_test]: Before sending goal
24: [behavior_server-4] [INFO] [1669705951.974123347] [behavior_server]: Running spin
24: [test_spin_behavior_fake_node-6] [INFO] [1669705951.974203773] [spin_behavior_test]: Waiting for result
24: [behavior_server-4] [INFO] [1669705951.974369486] [behavior_server]: Turning 3.14 for spin behavior.
24: [test_spin_behavior_fake_node-6] [INFO] [1669705952.974802649] [spin_behavior_test]: target_yaw 3.141593
24: [behavior_server-4] [WARN] [1669705953.074907474] [behavior_server]: Collision Ahead - Exiting Spin
24: [behavior_server-4] [WARN] [1669705953.075003019] [behavior_server]: spin failed
24: [behavior_server-4] [WARN] [1669705953.075168570] [behavior_server]: [spin] [ActionServer] Aborting handle.
24: [test_spin_behavior_fake_node-6] [INFO] [1669705953.140236205] [spin_behavior_test]: After sending goal
24: [test_spin_behavior_fake_node-6] [ERROR] [1669705953.140503623] [spin_behavior_test]: Goal was aborted
24: [test_spin_behavior_fake_node-6] [INFO] [1669705958.141105124] [spin_behavior_test]: Found current robot pose
24: [test_spin_behavior_fake_node-6] [INFO] [1669705958.141184045] [spin_behavior_test]: Init Yaw is 0.000000
24: [test_spin_behavior_fake_node-6] [INFO] [1669705958.141208443] [spin_behavior_test]: Before sending goal
24: [behavior_server-4] [INFO] [1669705958.242375494] [behavior_server]: Running spin
24: [behavior_server-4] [INFO] [1669705958.242588505] [behavior_server]: Turning 3.14 for spin behavior.
24: [test_spin_behavior_fake_node-6] [INFO] [1669705958.242471377] [spin_behavior_test]: Waiting for result
24: [test_spin_behavior_fake_node-6] [INFO] [1669705959.243126750] [spin_behavior_test]: target_yaw 3.141593
24: [behavior_server-4] [WARN] [1669705959.343066389] [behavior_server]: Collision Ahead - Exiting Spin
24: [behavior_server-4] [WARN] [1669705959.343138745] [behavior_server]: spin failed
24: [behavior_server-4] [WARN] [1669705959.343163801] [behavior_server]: [spin] [ActionServer] Aborting handle.
24: [test_spin_behavior_fake_node-6] [INFO] [1669705959.419282936] [spin_behavior_test]: After sending goal
24: [test_spin_behavior_fake_node-6] [ERROR] [1669705959.419900472] [spin_behavior_test]: Goal was aborted
24: [test_spin_behavior_fake_node-6] [       OK ] SpinRecoveryTests/SpinBehaviorTestFixture.testSpinRecovery/3141593_0100000 (18820 ms)
24: [test_spin_behavior_fake_node-6] [ RUN      ] SpinRecoveryTests/SpinBehaviorTestFixture.testSpinRecovery/4712389_0150000
24: [test_spin_behavior_fake_node-6] [INFO] [1669705964.420652504] [spin_behavior_test]: Found current robot pose
24: [test_spin_behavior_fake_node-6] [INFO] [1669705964.420747844] [spin_behavior_test]: Init Yaw is 0.000000
24: [test_spin_behavior_fake_node-6] [INFO] [1669705964.420776884] [spin_behavior_test]: Before sending goal
24: [behavior_server-4] [INFO] [1669705964.521998081] [behavior_server]: Running spin
24: [behavior_server-4] [INFO] [1669705964.522202383] [behavior_server]: Turning 4.71 for spin behavior.
24: [test_spin_behavior_fake_node-6] [INFO] [1669705964.522126288] [spin_behavior_test]: Waiting for result
24: [test_spin_behavior_fake_node-6] [INFO] [1669705965.522827297] [spin_behavior_test]: target_yaw 4.712389
24: [behavior_server-4] [WARN] [1669705965.622635436] [behavior_server]: Collision Ahead - Exiting Spin
24: [behavior_server-4] [WARN] [1669705965.622710701] [behavior_server]: spin failed
24: [behavior_server-4] [WARN] [1669705965.622738901] [behavior_server]: [spin] [ActionServer] Aborting handle.
24: [test_spin_behavior_fake_node-6] [INFO] [1669705965.695143348] [spin_behavior_test]: After sending goal
24: [test_spin_behavior_fake_node-6] [ERROR] [1669705965.695801477] [spin_behavior_test]: Goal was aborted
24: [test_spin_behavior_fake_node-6] [INFO] [1669705970.696493250] [spin_behavior_test]: Found current robot pose
24: [test_spin_behavior_fake_node-6] [INFO] [1669705970.696612092] [spin_behavior_test]: Init Yaw is 0.000000
24: [test_spin_behavior_fake_node-6] [INFO] [1669705970.696639233] [spin_behavior_test]: Before sending goal
24: [behavior_server-4] [INFO] [1669705970.797838758] [behavior_server]: Running spin
24: [behavior_server-4] [INFO] [1669705970.798044785] [behavior_server]: Turning 4.71 for spin behavior.
24: [test_spin_behavior_fake_node-6] [INFO] [1669705970.798035842] [spin_behavior_test]: Waiting for result
24: [test_spin_behavior_fake_node-6] [INFO] [1669705971.798688514] [spin_behavior_test]: target_yaw 4.712389
24: [behavior_server-4] [WARN] [1669705971.898530674] [behavior_server]: Collision Ahead - Exiting Spin
24: [behavior_server-4] [WARN] [1669705971.898601385] [behavior_server]: spin failed
24: [behavior_server-4] [WARN] [1669705971.898627562] [behavior_server]: [spin] [ActionServer] Aborting handle.
24: [test_spin_behavior_fake_node-6] [INFO] [1669705971.963030438] [spin_behavior_test]: After sending goal
24: [test_spin_behavior_fake_node-6] [ERROR] [1669705971.963199734] [spin_behavior_test]: Goal was aborted
24: [test_spin_behavior_fake_node-6] [INFO] [1669705976.963834141] [spin_behavior_test]: Found current robot pose
24: [test_spin_behavior_fake_node-6] [INFO] [1669705976.963912408] [spin_behavior_test]: Init Yaw is 0.000000
24: [test_spin_behavior_fake_node-6] [INFO] [1669705976.963936826] [spin_behavior_test]: Before sending goal
24: [behavior_server-4] [INFO] [1669705977.065053168] [behavior_server]: Running spin
24: [behavior_server-4] [INFO] [1669705977.065268398] [behavior_server]: Turning 4.71 for spin behavior.
24: [test_spin_behavior_fake_node-6] [INFO] [1669705977.065196865] [spin_behavior_test]: Waiting for result
24: [test_spin_behavior_fake_node-6] [INFO] [1669705978.065816212] [spin_behavior_test]: target_yaw 4.712389
24: [behavior_server-4] [WARN] [1669705978.165642011] [behavior_server]: Collision Ahead - Exiting Spin
24: [behavior_server-4] [WARN] [1669705978.165709108] [behavior_server]: spin failed
24: [behavior_server-4] [WARN] [1669705978.165741692] [behavior_server]: [spin] [ActionServer] Aborting handle.
24: [test_spin_behavior_fake_node-6] [INFO] [1669705978.227751237] [spin_behavior_test]: After sending goal
24: [test_spin_behavior_fake_node-6] [ERROR] [1669705978.227933375] [spin_behavior_test]: Goal was aborted
24: [test_spin_behavior_fake_node-6] [       OK ] SpinRecoveryTests/SpinBehaviorTestFixture.testSpinRecovery/4712389_0150000 (18808 ms)
24: [test_spin_behavior_fake_node-6] [ RUN      ] SpinRecoveryTests/SpinBehaviorTestFixture.testSpinRecovery/6283185_0100000
24: [test_spin_behavior_fake_node-6] [INFO] [1669705983.228664149] [spin_behavior_test]: Found current robot pose
24: [test_spin_behavior_fake_node-6] [INFO] [1669705983.228749236] [spin_behavior_test]: Init Yaw is 0.000000
24: [test_spin_behavior_fake_node-6] [INFO] [1669705983.228772504] [spin_behavior_test]: Before sending goal
24: [behavior_server-4] [INFO] [1669705983.329960556] [behavior_server]: Running spin
24: [behavior_server-4] [INFO] [1669705983.330159009] [behavior_server]: Turning -6.28 for spin behavior.
24: [test_spin_behavior_fake_node-6] [INFO] [1669705983.330102735] [spin_behavior_test]: Waiting for result
24: [test_spin_behavior_fake_node-6] [INFO] [1669705984.330778826] [spin_behavior_test]: target_yaw -6.283185
24: [behavior_server-4] [WARN] [1669705984.430577411] [behavior_server]: Collision Ahead - Exiting Spin
24: [behavior_server-4] [WARN] [1669705984.430646165] [behavior_server]: spin failed
24: [behavior_server-4] [WARN] [1669705984.430673498] [behavior_server]: [spin] [ActionServer] Aborting handle.
24: [test_spin_behavior_fake_node-6] [INFO] [1669705984.656678308] [spin_behavior_test]: After sending goal
24: [test_spin_behavior_fake_node-6] [ERROR] [1669705984.657253444] [spin_behavior_test]: Goal was aborted
24: [test_spin_behavior_fake_node-6] [INFO] [1669705989.657976595] [spin_behavior_test]: Found current robot pose
24: [test_spin_behavior_fake_node-6] [INFO] [1669705989.658054566] [spin_behavior_test]: Init Yaw is 0.000000
24: [test_spin_behavior_fake_node-6] [INFO] [1669705989.658078898] [spin_behavior_test]: Before sending goal
24: [behavior_server-4] [INFO] [1669705989.759238401] [behavior_server]: Running spin
24: [behavior_server-4] [INFO] [1669705989.759465737] [behavior_server]: Turning -6.28 for spin behavior.
24: [test_spin_behavior_fake_node-6] [INFO] [1669705989.759362432] [spin_behavior_test]: Waiting for result
24: [test_spin_behavior_fake_node-6] [INFO] [1669705990.759940939] [spin_behavior_test]: target_yaw -6.283185
24: [behavior_server-4] [WARN] [1669705990.859951210] [behavior_server]: Collision Ahead - Exiting Spin
24: [behavior_server-4] [WARN] [1669705990.860038501] [behavior_server]: spin failed
24: [behavior_server-4] [WARN] [1669705990.860079397] [behavior_server]: [spin] [ActionServer] Aborting handle.
24: [test_spin_behavior_fake_node-6] [INFO] [1669705991.117225855] [spin_behavior_test]: After sending goal
24: [test_spin_behavior_fake_node-6] [ERROR] [1669705991.117846877] [spin_behavior_test]: Goal was aborted
24: [test_spin_behavior_fake_node-6] [INFO] [1669705996.118706448] [spin_behavior_test]: Found current robot pose
24: [test_spin_behavior_fake_node-6] [INFO] [1669705996.118788952] [spin_behavior_test]: Init Yaw is 0.000000
24: [test_spin_behavior_fake_node-6] [INFO] [1669705996.118812080] [spin_behavior_test]: Before sending goal
24: [behavior_server-4] [INFO] [1669705996.219890824] [behavior_server]: Running spin
24: [behavior_server-4] [INFO] [1669705996.220082531] [behavior_server]: Turning -6.28 for spin behavior.
24: [test_spin_behavior_fake_node-6] [INFO] [1669705996.220107356] [spin_behavior_test]: Waiting for result
24: [test_spin_behavior_fake_node-6] [INFO] [1669705997.220786832] [spin_behavior_test]: target_yaw -6.283185
24: [behavior_server-4] [WARN] [1669705997.320623653] [behavior_server]: Collision Ahead - Exiting Spin
24: [behavior_server-4] [WARN] [1669705997.320717402] [behavior_server]: spin failed
24: [behavior_server-4] [WARN] [1669705997.320750855] [behavior_server]: [spin] [ActionServer] Aborting handle.
24: [test_spin_behavior_fake_node-6] [INFO] [1669705997.574991254] [spin_behavior_test]: After sending goal
24: [test_spin_behavior_fake_node-6] [ERROR] [1669705997.575923555] [spin_behavior_test]: Goal was aborted
24: [test_spin_behavior_fake_node-6] [       OK ] SpinRecoveryTests/SpinBehaviorTestFixture.testSpinRecovery/6283185_0100000 (19349 ms)
24: [test_spin_behavior_fake_node-6] [ RUN      ] SpinRecoveryTests/SpinBehaviorTestFixture.testSpinRecovery/12566371_0150000
24: [test_spin_behavior_fake_node-6] [INFO] [1669706002.576804016] [spin_behavior_test]: Found current robot pose
24: [test_spin_behavior_fake_node-6] [INFO] [1669706002.576883083] [spin_behavior_test]: Init Yaw is 0.000000
24: [test_spin_behavior_fake_node-6] [INFO] [1669706002.576908898] [spin_behavior_test]: Before sending goal
24: [behavior_server-4] [INFO] [1669706002.678191001] [behavior_server]: Running spin
24: [behavior_server-4] [INFO] [1669706002.678390567] [behavior_server]: Turning 12.57 for spin behavior.
24: [test_spin_behavior_fake_node-6] [INFO] [1669706002.678264673] [spin_behavior_test]: Waiting for result
24: [test_spin_behavior_fake_node-6] [INFO] [1669706003.678904893] [spin_behavior_test]: target_yaw 12.566371
24: [behavior_server-4] [WARN] [1669706003.778809258] [behavior_server]: Collision Ahead - Exiting Spin
24: [behavior_server-4] [WARN] [1669706003.778881143] [behavior_server]: spin failed
24: [behavior_server-4] [WARN] [1669706003.778906428] [behavior_server]: [spin] [ActionServer] Aborting handle.
24: [test_spin_behavior_fake_node-6] [INFO] [1669706004.142739363] [spin_behavior_test]: After sending goal
24: [test_spin_behavior_fake_node-6] [ERROR] [1669706004.143457171] [spin_behavior_test]: Goal was aborted
24: [test_spin_behavior_fake_node-6] [INFO] [1669706009.144418745] [spin_behavior_test]: Found current robot pose
24: [test_spin_behavior_fake_node-6] [INFO] [1669706009.144519085] [spin_behavior_test]: Init Yaw is 0.000000
24: [test_spin_behavior_fake_node-6] [INFO] [1669706009.144592775] [spin_behavior_test]: Before sending goal
24: [behavior_server-4] [INFO] [1669706009.245832486] [behavior_server]: Running spin
24: [behavior_server-4] [INFO] [1669706009.246036866] [behavior_server]: Turning 12.57 for spin behavior.
24: [test_spin_behavior_fake_node-6] [INFO] [1669706009.246026183] [spin_behavior_test]: Waiting for result
24: [test_spin_behavior_fake_node-6] [INFO] [1669706010.246547325] [spin_behavior_test]: target_yaw 12.566371
24: [behavior_server-4] [WARN] [1669706010.346410569] [behavior_server]: Collision Ahead - Exiting Spin
24: [behavior_server-4] [WARN] [1669706010.346482789] [behavior_server]: spin failed
24: [behavior_server-4] [WARN] [1669706010.346509751] [behavior_server]: [spin] [ActionServer] Aborting handle.
24: [test_spin_behavior_fake_node-6] [INFO] [1669706010.719520170] [spin_behavior_test]: After sending goal
24: [test_spin_behavior_fake_node-6] [ERROR] [1669706010.720089509] [spin_behavior_test]: Goal was aborted
24: [test_spin_behavior_fake_node-6] [INFO] [1669706015.720894596] [spin_behavior_test]: Found current robot pose
24: [test_spin_behavior_fake_node-6] [INFO] [1669706015.720970620] [spin_behavior_test]: Init Yaw is 0.000000
24: [test_spin_behavior_fake_node-6] [INFO] [1669706015.720992852] [spin_behavior_test]: Before sending goal
24: [behavior_server-4] [INFO] [1669706015.822168186] [behavior_server]: Running spin
24: [behavior_server-4] [INFO] [1669706015.822368401] [behavior_server]: Turning 12.57 for spin behavior.
24: [test_spin_behavior_fake_node-6] [INFO] [1669706015.822280995] [spin_behavior_test]: Waiting for result
24: [test_spin_behavior_fake_node-6] [INFO] [1669706016.822709576] [spin_behavior_test]: target_yaw 12.566371
24: [behavior_server-4] [WARN] [1669706016.922713433] [behavior_server]: Collision Ahead - Exiting Spin
24: [behavior_server-4] [WARN] [1669706016.922785707] [behavior_server]: spin failed
24: [behavior_server-4] [WARN] [1669706016.922814225] [behavior_server]: [spin] [ActionServer] Aborting handle.
24: [test_spin_behavior_fake_node-6] [INFO] [1669706017.295720086] [spin_behavior_test]: After sending goal
24: [test_spin_behavior_fake_node-6] [ERROR] [1669706017.296337844] [spin_behavior_test]: Goal was aborted
24: [test_spin_behavior_fake_node-6] [       OK ] SpinRecoveryTests/SpinBehaviorTestFixture.testSpinRecovery/12566371_0150000 (19720 ms)
24: [test_spin_behavior_fake_node-6] [----------] 7 tests from SpinRecoveryTests/SpinBehaviorTestFixture (95308 ms total)
24: [test_spin_behavior_fake_node-6] 
24: [test_spin_behavior_fake_node-6] [----------] Global test environment tear-down
24: [test_spin_behavior_fake_node-6] [==========] 9 tests from 2 test suites ran. (138098 ms total)
24: [test_spin_behavior_fake_node-6] [  PASSED  ] 9 tests.
24: [INFO] [test_spin_behavior_fake_node-6]: process has finished cleanly [pid 62677]
24: [INFO] [lifecycle_manager-5]: sending signal 'SIGINT' to process[lifecycle_manager-5]
24: [INFO] [behavior_server-4]: sending signal 'SIGINT' to process[behavior_server-4]
24: [INFO] [static_transform_publisher-3]: sending signal 'SIGINT' to process[static_transform_publisher-3]
24: [INFO] [static_transform_publisher-2]: sending signal 'SIGINT' to process[static_transform_publisher-2]
24: [INFO] [static_transform_publisher-1]: sending signal 'SIGINT' to process[static_transform_publisher-1]
24: [lifecycle_manager-5] [INFO] [1669706017.417682028] [rclcpp]: signal_handler(signum=2)
24: [behavior_server-4] [INFO] [1669706017.418357467] [rclcpp]: signal_handler(signum=2)
24: [static_transform_publisher-3] [INFO] [1669706017.419121061] [rclcpp]: signal_handler(signum=2)
24: [static_transform_publisher-2] [INFO] [1669706017.419853220] [rclcpp]: signal_handler(signum=2)
24: [static_transform_publisher-1] [INFO] [1669706017.420843119] [rclcpp]: signal_handler(signum=2)
24: [behavior_server-4] [INFO] [1669706017.426433013] [behavior_server]: Destroying
24: [behavior_server-4] terminate called after throwing an instance of 'rclcpp::exceptions::RCLError'
24: [behavior_server-4]   what():  Couldn't initialize rcl timer handle: the given context is not valid, either rcl_init() was not called or rcl_shutdown() was called., at ./src/rcl/guard_condition.c:67
24: [INFO] [lifecycle_manager-5]: process has finished cleanly [pid 62675]
24: [INFO] [static_transform_publisher-3]: process has finished cleanly [pid 62671]
24: [INFO] [static_transform_publisher-2]: process has finished cleanly [pid 62669]
24: [INFO] [static_transform_publisher-1]: process has finished cleanly [pid 62667]
24: [ERROR] [behavior_server-4]: process has died [pid 62673, exit code -6, cmd '/opt/overlay_ws/install/nav2_behaviors/lib/nav2_behaviors/behavior_server --ros-args -r __node:=behavior_server --params-file /tmp/tmpt5msp7p7 -r /tf:=tf -r /tf_static:=tf_static'].
24: -- run_test.py: return code 0
24: -- run_test.py: generate result file '/opt/overlay_ws/build/nav2_system_tests/test_results/nav2_system_tests/test_spin_behavior_fake.xml' with successful test
1/1 Test #24: test_spin_behavior_fake ..........   Passed  138.69 sec

The following tests passed:
	test_spin_behavior_fake

100% tests passed, 0 tests failed out of 1

```

**Review by SteveMacenski — APPROVED** [review]

**Comment by stevedanomodolor** [comment]
After the ci builds, I will into it and see if it takes into consideration the tests and add more test for full coverage for the time_behavior.cpp

**Comment by SteveMacenski** [comment]
I'm also not really sure what's up with that test failing. I took a quick look and it started failing about a week ago on merging the navigation piping PR from @jwallace42, but I don't think he did anything to cause it. Could be a rolling update sync that broke something. That would also be a nice thing to look at potentially if you wanted! If not, it'll be on my queue in the next week or so. 

```
/opt/overlay_ws/src/navigation2/nav2_behavior_tree/test/plugins/decorator/test_single_trigger_node.cpp:75
Expected equality of these values:
  bt_node_->executeTick()
    Which is: FAILURE
  BT::NodeStatus::RUNNING
    Which is: RUNNING/opt/overlay_ws/src/navigation2/nav2_behavior_tree/test/plugins/decorator/test_single_trigger_node.cpp:77
Expected equality of these values:
  bt_node_->executeTick()
    Which is: FAILURE
  BT::NodeStatus::SUCCESS
    Which is: SUCCESS
```

**Comment by stevedanomodolor** [comment]
I also noticed that too, I will look into it to see the whats the culprit

**Comment by SteveMacenski** [comment]
Thanks much! This is a big help!

**Pull request merged**